src/catkin_pkg/package.py

```diff
@@ -529,7 +540,7 @@ def parse_package_string(data, filename=None):
     if errors:
         raise InvalidPackage('Error(s) in %s:%s' % (filename, ''.join(['\n- %s' % e for e in errors])))
 
-    pkg.validate()
+    pkg.validate(warnings)
```

Comment:
Last comment - promise :wink: 
This one should use the keyword too: `warnings=warnings`.